### PR TITLE
feat: [CDS-52146]: Aws Lambda FF and file structure

### PR DESCRIPTION
**Pull request by rohitk-h**
## Describe your changes
- Introducing feature flag for AWS lambda implementation
- Added initial file structure and tasks. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43324)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTaskHelper.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+package io.harness.delegate.task.aws.lambda;
+
+import static io.harness.annotations.dev.HarnessTeam.CDP;
+
+import io.harness.annotations.dev.OwnedBy;
+
+import com.amazonaws.services.lambda.AWSLambda;
+import com.amazonaws.services.lambda.model.CreateFunctionRequest;
+import com.amazonaws.services.lambda.model.DeleteFunctionRequest;
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTaskHelper.java:16]: @rohitk-h Lets not use AWS SDK V1 classes. Lets use V2 SDK API classes.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public AwsLambdaCommandResponse run(Object[] parameters) {
+        throw new NotImplementedException("not implemented");
+    }
+
+    @Override
+    public AwsLambdaCommandResponse run(TaskParameters parameters) {
+        GoogleFunctionCommandRequest googleFunctionCommandRequest = (GoogleFunctionCommandRequest) parameters;
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java:42]: nit: Remove Google Functions.

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — CHANGES_REQUESTED** [review]
Please address review comments

**Comment by rohitk-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java` line 42 at r1](https://reviewable.io/reviews/harness/harness-core/43324#-NNAdRU650Rv8E4UDM9r-r1-42:-NNEOSqm7QJXXnTnv1M6:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/249ce09c148092df388786585782624618509fa6/930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java#L42)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

nit: Remove Google Functions.
</blockquote></details>

done

**Comment by rohitk-h** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTaskHelper.java` line 16 at r1](https://reviewable.io/reviews/harness/harness-core/43324#-NNAdRU650Rv8E4UDM9s-r1-16:-NNEOUj303cKI8_nSP_D:bvnrpcm) ([raw file](https://github.com/harness/harness-core/blob/249ce09c148092df388786585782624618509fa6/930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTaskHelper.java#L16)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

@rohitk-h Lets not use AWS SDK V1 classes. Lets use V2 SDK API classes.
</blockquote></details>

updated

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — COMMENTED** [review]
> Reviewed 3 of 11 files at r1, 18 of 18 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43324)* status: 21 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by sainath-b — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43324)* status: 21 file(s) to review. 1 of 2 approvals obtained.

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 3 of 11 files at r1, 18 of 18 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43324)* status: :shipit: complete! ✓

**Review by xingchi-jin — APPROVED** [review]
checked AbstractDelegateRunnableTask classes and capability overrides 

> Reviewed 3 of 11 files at r1, 18 of 18 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43324#-:-NNJYv9RBA_IxKjizMYu:b-r5q76j)* status: :shipit: complete! ✓

**Pull request merged**